File: kraken/src/main/java/glitch/kraken/object/json/Community.java
package glitch.kraken.object.json;

import glitch.api.objects.json.interfaces.IDObject;
import lombok.Data;

import java.util.Locale;
import java.util.UUID;

@Data
public class Community implements IDObject<UUID> {
    private final UUID id;
    private final Long ownerId;
    private final String name;
    private final String displayName;
    private final String avatarImageUrl;
    private final String coverImageUrl;
    private final String description;
    private final String descriptionHtml;
    private final String rules;
    private final String rulesHtml;
    private final Locale language;
    private final String summary;
}

File: kraken/src/main/kotlin/glitch/kraken/object/json/Community.kt
package glitch.kraken.`object`.json

import glitch.api.objects.json.interfaces.IDObject
import java.util.Locale
import java.util.UUID

/**
 *
 * @author Damian Staszewski [damian@stachuofficial.tv]
 * @version %I%, %G%
 * @since 1.0
 */
data class Community(
        override val id: UUID,
        val ownerId: Long,
        val name: String,
        val displayName: String,
        val avatarImageUrl: String,
        val coverImageUrl: String,
        val description: String,
        val descriptionHtml: String,
        val rules: String,
        val rulesHtml: String,
        val language: Locale,
        val summary: String
) : IDObject<UUID>
